cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -557,6 +562,12 @@ private void configureStreamInput(Job job, StreamBatchReadable stream) throws IO
 
     job.setInputFormatClass(StreamInputFormat.class);
 
+    try {
```

Comment:
Is this the only way that a MR can read a stream? Isn't there a way to specify it via ``setInputDataset("stream:<name>"``?